components/esp_http_client/esp_http_client.c

```diff
@@ -1155,6 +1155,18 @@ int esp_http_client_get_errno(esp_http_client_handle_t client)
     return esp_transport_get_errno(client->transport);
 }
 
+esp_err_t esp_http_client_get_tls_error(esp_http_client_handle_t client, int *tls_code, int *tls_flags){
+    if (!client || !tls_code || !tls_flags) {
+        ESP_LOGE(TAG, "Invalid client handle or params");
+        return ESP_FAIL;
+    }
+    return esp_tls_get_and_clear_last_error(
```

Comment:
Please have a look into the documentation for API esp_tls_get_and_clear_last_error() and see if all cases are covered correctly or not.

Reply:
changed params to be optionals as in the callee
```
esp_err_t esp_tls_get_and_clear_last_error(esp_tls_error_handle_t h, int *esp_tls_code, int *esp_tls_flags)
{
    if (!h) {
        return ESP_ERR_INVALID_STATE;
    }
    esp_err_t last_err = h->last_error;
    if (esp_tls_code) {
        *esp_tls_code = h->esp_tls_error_code;
    }
    if (esp_tls_flags) {
        *esp_tls_flags = h->esp_tls_flags;
    }
    memset(h, 0, sizeof(esp_tls_last_error_t));
    return last_err;
}
```